src/zenml/cli/stack.py

```diff
@@ -1250,3 +1278,316 @@ def register_secrets(
+    from mlstacks.utils import zenml_utils
+
+    cli_utils.warning(ALPHA_MESSAGE)
+
+    cli_params: Dict[str, Any] = ctx.params
+    stack, components = convert_click_params_to_mlstacks_primitives(cli_params)
+
+    cli_utils.declare("Checking flavor compatibility...")
+    if not zenml_utils.has_valid_flavor_combinations(stack, components):
+        cli_utils.error(
+            "The specified stack and component flavors are not compatible "
+            "with the provider or with one another. Please try again."
+        )
+
+    stack_dict, component_dicts = convert_mlstacks_primitives_to_dicts(
+        stack, components
+    )
+    # write the stack and component yaml files
+    from mlstacks.constants import MLSTACKS_PACKAGE_NAME
+
+    spec_dir = (
+        f"{click.get_app_dir(MLSTACKS_PACKAGE_NAME)}/stack_specs/{stack.name}"
+    )
+    cli_utils.declare(f"Writing spec files to {spec_dir}...")
+    create_dir_recursive_if_not_exists(spec_dir)
+
+    stack_file_path = f"{spec_dir}/stack-{stack.name}.yaml"
+    write_yaml(file_path=stack_file_path, contents=stack_dict)
+    for component in component_dicts:
+        write_yaml(
+            file_path=f"{spec_dir}/{component['name']}.yaml",
+            contents=component,
+        )
+
+    from mlstacks.utils import terraform_utils
+
+    cli_utils.declare("Deploying stack using Terraform...")
+    terraform_utils.deploy_stack(stack_file_path, debug_mode=debug_mode)
+    cli_utils.declare("Stack successfully deployed.")
+
+    if not no_import_stack_flag:
+        cli_utils.declare(f"Importing stack '{stack_name}' into ZenML...")
+        import_new_mlstacks_stack(
+            stack_name=stack_name,
+            provider=stack.provider,
+            stack_spec_dir=spec_dir,
+        )
+        cli_utils.declare("Stack successfully imported into ZenML.")
+
+
+@stack.command(
+    help="Destroy stack components created previously with "
+    "`zenml stack deploy`"
+)
+@click.argument("stack_name", required=True)
+@click.option(
+    "--provider",
+    "-p",
+    "provider",
+    type=click.Choice(STACK_RECIPE_MODULAR_RECIPES),
+    help="The cloud provider on which the stack is deployed.",
+)
+@click.option(
+    "--debug",
+    "-d",
+    "debug_mode",
+    is_flag=True,
+    default=False,
+    help="Whether to run Terraform in debug mode.",
+)
+def destroy(
+    stack_name: str,
+    provider: str,
+    debug_mode: bool = False,
+) -> None:
+    """Destroy all resources previously created with `zenml stack deploy`."""
+    cli_utils.verify_mlstacks_prerequisites_installation()
+    from mlstacks.constants import MLSTACKS_PACKAGE_NAME
+
+    # check the stack actually exists
+    if not stack_exists(stack_name):
+        cli_utils.error(
+            f"Stack with name '{stack_name}' does not exist. Please check and "
+            "try again."
+        )
+    spec_files_dir: str = (
+        f"{click.get_app_dir(MLSTACKS_PACKAGE_NAME)}/stack_specs/{stack_name}"
+    )
```

Comment:
`mlstacks` in its current implementation doesn't support this kind of multi-user
workflow as you suggest, mainly because we don't offer a remote state synced
between them. So basically it's 1 user on a single machine.